spec/request/service_offerings_spec.rb

```diff
@@ -67,7 +70,21 @@
         Hash.new(successful_response)
       end
 
-      it_behaves_like 'permissions for single object endpoint', ALL_PERMISSIONS
+      it_behaves_like 'permissions for single object endpoint', COMPLETE_PERMISSIONS
+
+      context 'when the hide_marketplace_from_unauthenticated_users feature flag is enabled' do
+        before do
+          VCAP::CloudController::FeatureFlag.create(name: 'hide_marketplace_from_unauthenticated_users', enabled: true)
+        end
+
+        let(:expected_codes_and_responses) do
+          h = Hash.new(successful_response)
+          h['unauthenticated'] = { code: 401 }
```

Comment:
Should this be a 401 or a 404?

Reply:
The V2 behaviour is to be 401, so it's consistent with V2 for this to be 401: https://github.com/cloudfoundry/cloud_controller_ng/blob/13fa61ceba7d731adb57c6dd3664f7c815875bd1/app/controllers/services/services_controller.rb#L30-L31